src/Bicep.LangServer/Handlers/BicepBuildParamsCommandHandler.cs

```diff
@@ -0,0 +1,120 @@
+{
+    // This handler is used to build compiled parameters.json file for given a bicep file path.
+    // It returns build-params succeeded/failed message, which can be displayed approriately in IDE output window
+    public class BicepBuildParamsCommandHandler : ExecuteTypedResponseCommandHandlerBase<string, string>
+    {
+        private readonly ICompilationManager compilationManager;
+        private readonly BicepCompiler bicepCompiler;
+
+        public BicepBuildParamsCommandHandler(ICompilationManager compilationManager, BicepCompiler bicepCompiler, ISerializer serializer)
+            : base(LangServerConstants.BuildParamsCommand, serializer)
+        {
+            this.compilationManager = compilationManager;
+            this.bicepCompiler = bicepCompiler;
+        }
+
+        public override async Task<string> Handle(string bicepParamsFilePath, CancellationToken cancellationToken)
+        {
+            if (string.IsNullOrWhiteSpace(bicepParamsFilePath))
+            {
+                throw new ArgumentException("Invalid input file path");
+            }
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepParamsFilePath);
+            string output = await GenerateCompiledParametersFileAndReturnOutputMessage(bicepParamsFilePath, documentUri);
+
+            return output;
+        }
+
+        private async Task<string> GenerateCompiledParametersFileAndReturnOutputMessage(string bicepParamsFilePath, DocumentUri documentUri)
+        {
+            string compiledFilePath = PathHelper.ResolveParametersFileOutputPath(bicepParamsFilePath, OutputFormatOption.Json);
+            string compiledFile = Path.GetFileName(compiledFilePath);
+
+            // If the template exists and contains bicep generator metadata, we can go ahead and replace the file.
+            // If not, we'll fail the generate params.
+            if (File.Exists(compiledFilePath) && !TemplateIsParametersFile(File.ReadAllText(compiledFilePath)))
+            {
+                return "Building parameters file failed. The file \"" + compiledFile + "\" already exists but does not contain the schema for a parameters file. If overwriting the file is intended, delete it manually and retry the Generate Parameters command.";
+            }
+
+            var compilation = await new CompilationHelper(bicepCompiler, compilationManager).GetRefreshedCompilation(documentUri);
+            var fileUri = documentUri.ToUri();
+
+            var diagnosticsByFile = compilation.GetAllDiagnosticsByBicepFile().FirstOrDefault(x => x.Key.FileUri == fileUri);
+
+            if (diagnosticsByFile.Value.Any(x => x.Level == DiagnosticLevel.Error))
+            {
+                return "Building parameters file failed. Please fix below errors:\n" + DiagnosticsHelper.GetDiagnosticsMessage(diagnosticsByFile);
+            }
+
+            var existingContent = File.Exists(compiledFilePath) ? File.ReadAllText(compiledFilePath) : string.Empty;
```

Comment:
Is this variable used below? If not, can you remove it?

Reply:
good catch, give me a minute. I just re-run the tests..

Reply:
tests succeeded 👍 created commit to remove the unused variable. thanks 👍